File: 0088-merge-sorted-array/0088-merge-sorted-array.js
/**
 * @param {number[]} nums1
 * @param {number} m
 * @param {number[]} nums2
 * @param {number} n
 * @return {void} Do not return anything, modify nums1 in-place instead.
 */
var merge = function(nums1, m, nums2, n) {
    let i = m-1;
    let j = m+n-1;
    let k = n-1;
    while (k>=0){
        while (nums1[i] >= nums2[k]){
            nums1[j] = nums1[i];
            j--;
            i--;
        }
        nums1[j] = nums2[k];
        j--;
        k--;
    }
    
};

File: 0088-merge-sorted-array/0088-merge-sorted-array.py
class Solution(object):
    def merge(self, nums1, m, nums2, n):
        """
        :type nums1: List[int]
        :type m: int
        :type nums2: List[int]
        :type n: int
        :rtype: None Do not return anything, modify nums1 in-place instead.
        """
        nums3 = nums1[:]
        k, j = 0, 0
        for i in range(m+n):
            if j >= m or (k < n and nums3[j] > nums2[k]):
                nums1[i] = nums2[k]
                k += 1
            else:
                nums1[i] = nums3[j]
                j += 1
        